### PR TITLE
Allow passing `config_file` argument to ControlNetModel when using `from_single_file`

### PR DESCRIPTION
**Pull request by DN6**
# What does this PR do?
Similar fix to : #6950  

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/diffusers/loaders/controlnet.py [hunk 1]
```diff
@@ -38,6 +38,9 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
                     - A link to the `.ckpt` file (for example
                       `"https://huggingface.co/<repo_id>/blob/main/<path_to_file>.ckpt"`) on the Hub.
                     - A path to a *file* containing all pipeline weights.
+            config_file (`str`, *optional*):
+                Filepath to the configuration YAML file associated with the model. If not provided it will default to:
+                https://raw.githubusercontent.com/lllyasviel/ControlNet/main/models/cldm_v15.yaml
```
Comment (sayakpaul) [inline src/diffusers/loaders/controlnet.py:43]: I don't think it makes the difference between `original_config_file` and `config_file` quite clear. Could you rephrase?

Comment (DN6) [reply src/diffusers/loaders/controlnet.py:43]: They both mean the same thing. We used `original_config_file` in the pipeline Mixin. 

With the refactor, the internal logic `config_file = kwargs.pop("config_file")` was removed in favour of using `original_config_file` like we do for the Pipelines. 

But UI's like SD.Next use `config_file` when loading the VAE.  Putting it back here again for backwards compatibility. 

We don't need to keep both. We can deprecate one of them.

Comment (sayakpaul) [reply src/diffusers/loaders/controlnet.py:43]: Okay. Then maybe in a follow-up PR, let's start their deprecation cycles?

**Review by sayakpaul — APPROVED** [review]
